### PR TITLE
Feat: Product listing page and Authentication

### PR DESCRIPTION
**Pull request by ramalingaa**
**Product listing page is added**
  - Add/remove to cart fun is added
  - Add/remove to wish list fun is added
-Increase/decrease cart items fun is added

**Login, signup and reset password pages are added**
  - User can now signup and sign in to access private routes

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -329,4 +329,279 @@ a {
...
+}
+.product-container {
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  row-gap: 1rem;
+  padding: 1rem 0 0 0 ;
+}
+.products-number-title {
+  padding: 1rem 0 ;
+  /* font-weight: bold; */
```
Comment (anuj-kumary) [inline src/App.css]: Remove unwanted comments

Comment (Rohanmond) [inline src/App.css]: remove this unwanted comment

File: src/frontend/components/Authentication/ForgotPassword.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { useAuthContext } from '../../context/context-index'
+import { Link, useNavigate } from "react-router-dom"
+
+
+export default function ForgotPassword() {
+    const [showPassword, setShowPassword] = useState({password:false, reEnterPasswrod:false})
+    const [isPasswordMatch, setISPasswordMatch] = useState(true)
+    const [newUserData, setNewUserData] = useState({ password:""})
+
+    const { jwtToken, setJwtToken } = useAuthContext()
```
Comment (anuj-kumary) [inline src/frontend/components/Authentication/ForgotPassword.jsx]: Here you're not using jwtToken so no need to import useAuthContext

File: src/frontend/components/Home/Category.jsx [hunk 3]
```diff
@@ -4,31 +4,31 @@ import { Link } from 'react-router-dom';
...
         </Link>
-        <Link to = "./Product" className="cat-card-wrapper">
+        <Link to = "./products" className="cat-card-wrapper">
             <div>
                 <img src = "https://media.istockphoto.com/photos/happy-curly-boy-laughing-and-holding-skateboard-picture-id1124742827?b=1&k=20&m=1124742827&s=170667a&w=0&h=FjLMBTxiZOhJB45V-L3pm_tYC4dYqHyTD2e_nRJN5LE=" alt = "Men Category" className="cat-card-img"/>
                 <p>Kids</p>
             </div>
         </Link>
-        <Link to = "./Product" className="cat-card-wrapper">
+        <Link to = "./products" className="cat-card-wrapper">
```
Comment (Rohanmond) [inline src/frontend/components/Home/Category.jsx:25]: better to use useNavigate herer

**Review by Rohanmond — APPROVED** [review]

**Review by anuj-kumary — APPROVED** [review]

**Pull request merged**